### PR TITLE
Add source ddl classes

### PR DESCRIPTION
**Pull request by Deep1998**
Create classes similar to spanner ddl but for source databases which allows storing the schema for migrations. This would be useful to remove dependency on session file for identifying source types.

### CODE REVIEW

File: v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/sourceddl/SourceInformationSchemaScanner.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.sourceddl;
+
+/**
+ * Interface for scanning information schema of source databases. Different implementations will
+ * handle different source database types.
+ */
+public interface SourceInformationSchemaScanner {
```
Comment (bharadwaj-aditya) [inline v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/sourceddl/SourceInformationSchemaScanner.java]: nit: This should probably just be sourceSchemaScanner.

Depending on the database, it would figure it if needs to get it from info schema or some other place.

File: v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/sourceddl/SourceSchema.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public abstract Builder sourceType(SourceDatabaseType sourceType);
+
+    public abstract Builder tables(ImmutableMap<String, SourceTable> tables);
+
+    public abstract Builder version(String version);
+
+    public abstract SourceSchema build();
+  }
+
+  public SourceTable table(String tableName) {
```
Comment (bharadwaj-aditya) [inline v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/sourceddl/SourceSchema.java:58]: wondering if having this method makes sense.  This can as well be an interface.

**Review by bharadwaj-aditya — COMMENTED** [review]
Some of these abstract classes seem like shell implementations. They can be interfaces as well to simplify the structure.

Please add some test cases with edge cases , null handling, missing fields etc.

File: v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/sourceddl/SourceSchema.java (hunk 2 above)
Comment (Deep1998) [reply v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/sourceddl/SourceSchema.java:58]: This is emulating the spanner DDL classes we have. It is a helper method to make it easier to access tables in the code which we do a lot. It otherwise tables 2 method calls. And moving this class to an interface would lead to a lot of boilerplate code which we are able to avoid using autovalue.

**Comment by Deep1998** [comment]
> Some of these abstract classes seem like shell implementations. They can be interfaces as well to simplify the structure.
> 
> Please add some test cases with edge cases , null handling, missing fields etc.

Thanks, added more test cases and removed the AbstractSourceInformationSchemaScanner for simplicity. The data model classes (SourceSchema, SourceTable, SourceColumn) are abstract classes because they use AutoValue, which allows us to avoid a lot of boiler plate code, so those I have kept as is.
